star-ui/src/components/BatteryPanel.tsx

```
@@ -1,60 +1,110 @@
+const getBatteryColor = (soc: number) => {
+  if (soc > 50) return '#30D158'; // System Green
+  if (soc > 20) return '#FF9F0A'; // System Orange
+  return '#FF453A'; // System Red
+};
+
+const SegmentedBatteryIcon = ({ soc, dimmed = false }: { soc: number; dimmed?: boolean }) => {
+  const color = dimmed ? 'rgba(255,255,255,0.1)' : getBatteryColor(soc);
+  const segments = dimmed ? 0 : Math.max(1, Math.ceil(soc / 20)); // 1 to 5 segments, 0 if dimmed
+
+  return (
+    <svg width="64" height="32" viewBox="0 0 64 32" fill="none" xmlns="http://www.w3.org/2000/svg" style={{ filter: dimmed ? 'none' : `drop-shadow(0 0 8px ${color}66)` }}>
```

Biome (2.4.4)

[error] Alternative text title element cannot be empty

(lint/a11y/noSvgWithoutTitle)